server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -718,18 +715,18 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
         try (BytesStreamOutput bStream = new BytesStreamOutput()) {
             try (StreamOutput stream = new OutputStreamStreamOutput(bStream)) {
                 XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON, stream);
-                repositoryData.snapshotsToXContent(builder, ToXContent.EMPTY_PARAMS);
+                repositoryData.snapshotsToXContent(builder);
                 builder.close();
             }
             snapshotsBytes = bStream.bytes();
         }
         // write the index file
-        final String indexBlob = INDEX_FILE_PREFIX + Long.toString(newGen);
+        final String indexBlob = INDEX_FILE_PREFIX + newGen;
         logger.debug("Repository [{}] writing new index generational blob [{}]", metadata.name(), indexBlob);
         writeAtomic(indexBlob, snapshotsBytes, true);
         // delete the N-2 index file if it exists, keep the previous one around as a backup
         if (isReadOnly() == false && newGen - 2 >= 0) {
-            final String oldSnapshotIndexFile = INDEX_FILE_PREFIX + Long.toString(newGen - 2);
+            final String oldSnapshotIndexFile = INDEX_FILE_PREFIX + (newGen - 2);
```

Comment:
I like the previous version better because here `+` stands for string concatenation and `-` stand for arithmetics. Explicit conversion makes it easier to read.

Reply:
sure why not, reverted these :)